source/join.d.ts

```diff
@@ -1,22 +1,22 @@
 /**
-Join an array of strings using the given string as delimiter.
+Join an array of strings and/or numbers using the given string as delimiter.
 
 Use-case: Defining key paths in a nested object. For example, for dot-notation fields in MongoDB queries.
 
 @example
 ```
 import {Join} from 'type-fest';
 
-const path: Join<['foo', 'bar', 'baz'], '.'> = ['foo', 'bar', 'baz'].join('.');
+const path: Join<['foo', '0', 'baz'], '.'> = ['foo', '0', 'baz'].join('.');
```

Comment:
Shouldn't it use number literals here?

Reply:
fixed here: https://github.com/sindresorhus/type-fest/pull/258/commits/ddf70235718d2e7753be483a23a385abdfdabfb0#diff-665c8ccce5a5adf0c05f3a4fc2ecdb0e16ca90812a3df2bd54fb05eed1ed94c7R11